src/test/fuzz/txreconciliation.cpp

```diff
@@ -0,0 +1,126 @@
+        Wtxid random_wtxid = Wtxid::FromUint256(ConsumeUInt256(fuzzed_data_provider));
+
+        CallOneOf(
+            fuzzed_data_provider,
+            [&] {
+                auto result = txrecontracker.AddToSet(rand_peer, random_wtxid);
+                if (result.m_succeeded) {
+                    assert(txrecontracker.IsTransactionInSet(rand_peer, random_wtxid, true));
+                    peer_set_sizes[rand_peer_index]++;
+                } else if (result.m_conflict.has_value()) {
+                    Wtxid collision;
+                    uint32_t short_id;
+                    assert(txrecontracker.HasCollision(rand_peer, random_wtxid, collision, short_id));
+                    assert(collision == result.m_conflict.value());
+                }
+            },
+            [&] {
+                (void)txrecontracker.ReadyDelayedTransactions(rand_peer);
+            },
+            [&] {
+                bool include_delayed = fuzzed_data_provider.ConsumeBool();
+                (void)txrecontracker.IsTransactionInSet(rand_peer, random_wtxid, include_delayed);
+            },
+            [&] {
+                bool result = txrecontracker.TryRemovingFromSet(rand_peer, random_wtxid);
+                if (result) {
+                    assert(!txrecontracker.IsTransactionInSet(rand_peer, random_wtxid, true));
+                    if (peer_set_sizes[rand_peer_index] > 0) {
+                        peer_set_sizes[rand_peer_index]--;
+                    }
+                }
+            },
+            [&] {
+                txrecontracker.ForgetPeer(rand_peer);
+                registered_peers.erase(registered_peers.begin() + rand_peer_index);
+                peer_set_sizes[rand_peer_index] = 0;
+                assert(!txrecontracker.IsPeerRegistered(rand_peer));
+                assert(!txrecontracker.IsTransactionInSet(rand_peer, random_wtxid, true));
+                pre_registered_peers.erase(rand_peer);
+            },
+            [&] {
+                assert(txrecontracker.IsPeerRegistered(rand_peer));
+            },
+            [&] {
+                size_t inbounds_fanout = fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, NUM_PEERS);
+                size_t outbounds_fanout = fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, NUM_PEERS);
+                auto fanout_targets = txrecontracker.GetFanoutTargets(random_wtxid, inbounds_fanout, outbounds_fanout);
+                (void)txrecontracker.ShouldFanoutTo(rand_peer, fanout_targets);
+            },
+            [&] {
+                std::vector<Wtxid> parents;
+                int num_parents = fuzzed_data_provider.ConsumeIntegralInRange<int>(0, 10);
+                for (int i = 0; i < num_parents; ++i) {
+                    parents.push_back(Wtxid::FromUint256(ConsumeUInt256(fuzzed_data_provider)));
+                }
+                auto sorted_peers = txrecontracker.SortPeersByFewestParents(parents);
```

Comment:
If we don't intend to use `sorted_peers`, we can simply do: `(void)txrecontracker.SortPeersByFewestParents(parents)`

Reply:
I had it because I was planning to check that the peers are actually sorted by fewest parents. Let me add it and see what you think. Might be a good check?